components/02-molecules/banner/grand-hero/yds-grand-hero.js

```diff
@@ -0,0 +1,8 @@
+Drupal.behaviors.grandHero = {
+  attach() {
+    const header = document.querySelector('.site-header');
+    document
+      .getElementsByClassName('grand-hero-banner')[0]
+      .style.setProperty('--site-header-height', header.offsetHeight);
+  },
+};
```

Comment:
This JS didn't work for me. I had to update it to this in order to get it working. You can test it on http://localhost:6006/?path=/story/page-examples-standard-pages--with-banner&globals=outline:true

```
Drupal.behaviors.grandHero = {
  attach() {
    const header = document.querySelector('.site-header');

    window.addEventListener('load', () => {
      const grandHeroes = document.querySelectorAll('.grand-hero-banner');

      grandHeroes.forEach((hero) => {
        hero.style.setProperty(
          '--site-header-height',
          `${header.offsetHeight}px`,
        );
      });
    });
  },
};
```

Reply:
However, I don't think this is the right way to do it. I didn't finish my thought yesterday, so I'll try and write it out here:

- [ ] Since we're setting a variable called `--site-header-height` I think that should happen as a part of the `site-header` component. So, I think we can move this JS file into the `site-header` directory. (and rename the behavior to match.)

- [ ] Then, we should set the initial value of the variable in the `04-page-layouts/page-layouts.scss` file. something like:
```
body {
  --site-header-height: 4.8125rem; // 77px / 16
}
```
This way, any future component that needs to react based on the header height also has access to the variable. If it's inside the `.grand-hero-banner` scope, then no other component can access the value, and we'll have to do more js in order for them to have the same value. It'd be better to do it at a high level - since we're doing it anyway, let's just do it once so that this component, and any future ones can all use the same value.

- [ ] Then, we can just set the variable at the "body" scope in JS like this:
```
Drupal.behaviors.siteHeader = {
  attach(context) {
    const body = context.querySelector('body');
    const header = context.querySelector('.site-header');

    window.addEventListener('load', () => {
      body.style.setProperty(
        '--site-header-height',
        `${header.offsetHeight}px`,
      );
    });
  },
};
```

Reply:
Oh, also, this doesn't adjust when the window resizes. Any time we scale the browser, we should recalculate it. e.g. when the menu wraps to a new line, or on mobile.

I messed with this a little tonight, and got close, but it's not quite working... This should get you most of the way there thought!

```
Drupal.behaviors.siteHeader = {
  attach(context) {
    const body = context.querySelector('body');
    const header = context.querySelector('.site-header');

    /**
     * debounce
     * @description Debounce to only run a function at most once every 200ms.
     * @param {} func The function to be run after the timeout.
     */
    const debounce = (func) => {
      let timer;
      return function debounceFunction(event) {
        if (timer) clearTimeout(timer);
        timer = setTimeout(func, 200, event);
      };
    };

    /**
     * setHeaderHeight
     * @description Set the `--site-header-height` variable.
     */
    const setHeaderHeight = () => {
      console.log(`${header.offsetHeight}px`);
      body.style.setProperty(
        '--site-header-height',
        `${header.offsetHeight}px`,
      );
    };

    // Determine, and set the site header height variable on page load.
    window.addEventListener('load', () => {
      setHeaderHeight();
    });

    // Update the site header height variable when the window is resized.
    window.addEventListener(
      'resize',
      debounce(function runSetHeight() {
        setHeaderHeight();
      }),
    );
  },
};
```